File: src/main/java/abstract_factory_java/factory/impl/ProgrammerPropertyFactory.java
package abstract_factory_java.factory.impl;

import abstract_factory_java.factory.PropertyFactory;
import abstract_factory_java.property.Car;
import abstract_factory_java.property.House;
import abstract_factory_java.property.Money;
import abstract_factory_java.property.impl.programmer.ProgrammerCar;
import abstract_factory_java.property.impl.programmer.ProgrammerHouse;
import abstract_factory_java.property.impl.programmer.ProgrammerMoney;

/**
 * @author angus
 * @date 2019/11/21
 */
public class ProgrammerPropertyFactory implements PropertyFactory {
    @Override
    public Car buyCar() {
        return new ProgrammerCar();
    }

    @Override
    public House buyHouse() {
        return new ProgrammerHouse();
    }

    @Override
    public Money drawMoney() {
        return new ProgrammerMoney();
    }
}

File: src/main/kotlin/abstract_factory/factory/ProgrammerPropertyFactory.kt
package abstract_factory.factory.impl

import abstract_factory.factory.PropertyFactory
import abstract_factory.property.Car
import abstract_factory.property.House
import abstract_factory.property.Money
import abstract_factory.property.impl.programmer.ProgrammerCar
import abstract_factory.property.impl.programmer.ProgrammerHouse
import abstract_factory.property.impl.programmer.ProgrammerMoney

class ProgrammerPropertyFactory : PropertyFactory {
    override fun buyCar(): Car = ProgrammerCar()

    override fun buyHouse(): House = ProgrammerHouse()

    override fun drawMoney(): Money = ProgrammerMoney()
}